lib/src/StatusNotification/StatusOSNotification.cpp

```diff
@@ -0,0 +1,123 @@
+
+static const UINT NOTIFYICONID = 0;
+
+static std::pair<HWND, StatusOSNotification *> HWND_INSTANCE_PAIR;
+#endif
+
+StatusOSNotification::StatusOSNotification(QObject *parent)
+    : QObject(parent)
+{
+#ifdef Q_OS_WIN
+    m_hwnd = nullptr;
+    initNotificationWin();
+#elif defined Q_OS_MACOS
+    m_notificationHelper = nullptr;
+    initNotificationMacOs();
+#endif
+}
+
+#ifdef Q_OS_WIN
+LRESULT CALLBACK StatusWndProc(HWND hwnd, UINT uMsg, WPARAM wParam, LPARAM lParam)
+{
+    const int msgInfo = LOWORD(lParam);
+
+    if (hwnd == HWND_INSTANCE_PAIR.first 
+        && HWND_INSTANCE_PAIR.second 
+        && HWND_INSTANCE_PAIR.second->m_identifiers.contains(uMsg) 
+        && msgInfo == NIN_BALLOONUSERCLICK)
+    {
+        HWND_INSTANCE_PAIR.second->notificationClicked(
+            HWND_INSTANCE_PAIR.second->m_identifiers[uMsg]);
+        return 0;
+    }
+
+    return DefWindowProc(hwnd, uMsg, wParam, lParam);
+}
+
+bool StatusOSNotification::initNotificationWin()
+{
+    // m_hwnd should be init only once, but that's would be a case if we system 
+    // tray window from here. But since we already have system tray added in the
+    // app we're just refering to that already added window and listen for events
+    // on it. HWND of that window may be changed during the runtime and we don't
+    // have an option to be notified about that, that's why we're searching for 
+    // appropriate HWND each time we need it, and that's why the following two
+    // lines are commented out.
+    //
+    // if (m_hwnd)
+    //     return true;
+
+    const QString cName = "QTrayIconMessageWindowClass";
+    LPCSTR className = cName.toStdString().c_str();
+    const QString wName = "QTrayIconMessageWindow";
+    LPCSTR windowName = wName.toStdString().c_str();
+
+    const auto appInstance = static_cast<HINSTANCE>(GetModuleHandle(nullptr));
+
+    WNDCLASSEX wc;
+    wc.cbSize = sizeof(WNDCLASSEX);
+    wc.style = CS_HREDRAW | CS_VREDRAW;
+    wc.lpfnWndProc = StatusWndProc;
+    wc.cbClsExtra = 0;
+    wc.cbWndExtra = 0;
+    wc.hInstance = appInstance;
+    wc.hCursor = nullptr;
+    wc.hbrBackground = (HBRUSH)(COLOR_WINDOW);
+    wc.hIcon = nullptr;
+    wc.hIconSm = nullptr;
+    wc.lpszMenuName = nullptr;
+    wc.lpszClassName = className;
+
+    ATOM atom = RegisterClassEx(&wc);
+    if (!atom)
+        printf("StatusOsNotification registering window class failed.\n");
+
+    m_hwnd = FindWindowExA(0, 0, className, windowName);
+    HWND_INSTANCE_PAIR = std::make_pair(m_hwnd, this);
+    return true;
+}
+#endif
+
+void StatusOSNotification::showNotification(QString title, QString message,
+                                            QString identifier)
+{
+#ifdef Q_OS_WIN
+    if (!initNotificationWin())
+    {
+        return;
+    }
+
+    NOTIFYICONDATA tnd;
+    memset(&tnd, 0, sizeof(NOTIFYICONDATA));
+    tnd.cbSize = sizeof(NOTIFYICONDATA);
+    tnd.uVersion = NOTIFYICON_VERSION_4;
+
+    const char *cTitle = title.toLocal8Bit().data();
```

Comment:
`toLocal8Bit()`:
`The returned byte array is undefined if the string contains characters not supported by the local 8-bit encoding.`
would this work for translated strings? maybe it worth considering unicode version if it exists?